builtin/plugins/user_deployer_whitelist/user_deployer_whitelist.go

```diff
@@ -260,11 +260,21 @@ func (uw *UserDeployerWhitelist) RemoveUserDeployer(ctx contract.Context, req *u
 	if err := contract.CallMethod(ctx, dwAddr, "RemoveUserDeployer", removeUserDeployerRequest, nil); err != nil {
 		return errors.Wrap(err, "failed to remove deployer")
 	}
-	// remove from current contract deployer address
-	if !ctx.Has(deployerStateKey(deployerAddr)) {
-		return ErrDeployerDoesNotExist
+	if ctx.FeatureEnabled(loomchain.UserDeployerWhitelistVersion1_1Feature, false) {
+		var userDeployer UserDeployerState
+		if err := ctx.Get(deployerStateKey(deployerAddr), &userDeployer); err != nil {
+			return errors.Wrap(err, "Failed to Get Deployer State")
+		}
+		userDeployer.Inactive = true
+		if err := ctx.Set(deployerStateKey(deployerAddr), &userDeployer); err != nil {
+			return errors.Wrap(err, "Saving WhitelistedDeployer in whitelisted deployers state")
+		}
+	} else {
```

Comment:
Add `return nil` on the line above this one, and then you won't need to nest the code below within this `else` block.

Reply:
done